behaviour/graql/explanation/reasoner.feature

```diff
@@ -0,0 +1,502 @@
+      (superior: $l, subordinate: $k) isa location-hierarchy;
+      get;
+      """
+
+    Then concept identifiers are
+      |     | check | value             |
+      | KC  | key   | name:King's Cross |
+      | UK  | key   | name:UK           |
+      | LDN | key   | name:London       |
+      | KCN | value | name:King's Cross |
+
+    Then uniquely identify answer concepts
+      | k  | l   | n   |
+      | KC | UK  | KCN |
+      | KC | LDN | KCN |
+
+    Then rules are
+      |                                 | when                                                                                                                 | then                                                         |
+      | location-hierarchy-transitivity | { (superior: $a, subordinate: $b) isa location-hierarchy; (superior: $b, subordinate: $c) isa location-hierarchy; }; | { (superior: $a, subordinate: $c) isa location-hierarchy; }; |
+
+    Then answers contain explanation tree
+      |   | children | vars    | identifiers | rule                            | pattern                                                                                                                                                                                          |
+      | 0 | 1, 2     | k, l, n | KC, UK, KCN | join                            | { $k isa area; $k has name $n; (superior: $l, subordinate: $k) isa location-hierarchy; $k id <answer.k.id>; $n id <answer.n.id>; $l id <answer.l.id>; };                                         |
+      | 1 | -        | k, n    | KC, KCN     | lookup                          | { $k isa area; $k has name $n; $n id <answer.n.id>; $k id <answer.k.id>; };                                                                                                                      |
+      | 2 | 3        | k, l    | KC, UK      | location-hierarchy-transitivity | { (superior: $l, subordinate: $k) isa location-hierarchy; $k isa area; $k id <answer.k.id>; $l id <answer.l.id>; };                                                                              |
+      | 3 | 4, 5     | a, b, c | UK, LDN, KC | join                            | { (superior: $a, subordinate: $b) isa location-hierarchy; (superior: $b, subordinate: $c) isa location-hierarchy; $c isa area; $a id <answer.a.id>; $b id <answer.b.id>; $c id <answer.c.id>; }; |
+      | 4 | -        | b, c    | LDN, KC     | lookup                          | { (superior: $b, subordinate: $c) isa location-hierarchy; $c isa area; $b id <answer.b.id>; $c id <answer.c.id>; };                                                                              |
+      | 5 | -        | a, b    | UK, LDN     | lookup                          | { (superior: $a, subordinate: $b) isa location-hierarchy; $b id <answer.b.id>; $a id <answer.a.id>; };                                                                                           |
+
+#   TODO Non-deterministically getting this error:
```

Comment:
is this still a valid TODO?

Reply:
I haven't had it fail for a long time, but I think it is/may still be possible that reasoner appends extra elements to the pattern. In this case is uses `area` when resolving and could add it in. I want to look out for this. Perhaps the TODO  can go until it's proven again to be non-deterministic.